### ISSUE TITLE
CSS parser throws CSS_SYNTAX_ERROR on comments

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-validator#361]
From https://github.com/mozilla/addons-validator/issues/347#issuecomment-160486165

This CSS causes a syntax error in the parser.

```css
#box /*.main{float:left;*/{width:auto !important;}
```

I suspect this might be a bug in the parser rather than our code because we're already skipping over comments. It might be due to the presence of `{` in the comment or that the comment is after the selector but before the opening `{`.

**Comment by minj** [comment]
```css
#box /*.main{float:left;}*/{width:auto !important;}
```
works fine, the key problem is unmatched braces

**Comment by muffinresearch** [comment]
Hmm, yeah it's also a bit more than that as the parser seems to make a mess of the selectors with a comment in between the selector and the opening `{`. See http://iamdustan.com/reworkcss_ast_explorer/ with this: `#box /*.main{float:left;}*/ {width:auto !important;}`

**Comment by muffinresearch** [comment]
Seems to show `#box /*.main` as a selector and `*/` as another one :/

**Comment by minj** [comment]
Indeed, the comments should be stripped first during AST generation

**Comment by muffinresearch** [comment]
This will be resolved by #364
